src/nvim/drawscreen.c

```diff
@@ -92,125 +89,279 @@ typedef enum {
   WC_BOTTOM_RIGHT,
 } WindowCorner;
 
-/// Whether to call "ui_call_grid_resize" in win_grid_alloc
-static bool send_grid_resize = false;
-
 #ifdef INCLUDE_GENERATED_DECLARATIONS
 # include "drawscreen.c.generated.h"
 #endif
 
+static bool redraw_popupmenu = false;
+static bool msg_grid_invalid = false;
+static bool resizing = false;
+
 static char *provider_err = NULL;
 
-/// Redraw a window later, with update_screen(type).
+/// Check if the cursor line needs to be redrawn because of 'concealcursor'.
 ///
-/// Set must_redraw only if not already set to a higher value.
-/// e.g. if must_redraw is CLEAR, type NOT_VALID will do nothing.
-void redraw_later(win_T *wp, int type)
-  FUNC_ATTR_NONNULL_ALL
+/// When cursor is moved at the same time, both lines will be redrawn regardless.
+void conceal_check_cursor_line(void)
```

Comment:
In vim this is defined in `screen.c` which is where `conceal_cursor_used` was also defined.

Would it be better to move this to `drawline.c`? Otherwise, I think this should be moved back to `screen.c`

Reply:
I moved it here because it calls `redrawWinline()`, whereas in Vim it doesn't.

Reply:
Eek, then there must be a bit of circular dependency going on. `screen.c` shouldn't have visibility to `drawscreen.c`.